### DISCUSSION TITLE
Okuldaki dolaplar sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#163] [Soru Cevap]
Bir okuldaki 100 kapalı dolabın önünde birer öğrenci duracak şekilde yerleşirler.
1. düdük çaldığında 1 in katları olan tüm dolap kapakları açılır.
2. düdük çaldığında 2 nin katları olan dolaplar açıksa kapanır kapalıysa açılır,
3. düdük çaldığında 3 ün katları olan dolaplar yine aynı şekilde kapanır veya açılır,
. . .
Bu böylece 100. düdük çalana kadar devam eder. 100. düdük çaldığında sadece 100. dolap kapağı açıksa kapanır kapalıysa açılır.
Soru şu, bu işlemin sonucunda kaç dolap kapağı açık kalır? Hangileridir?

**Comment by tolga0102** [comment]
```python
açık=[]
for a in range(1,101):
    b=0
    for c in range(1,101):
        if a%c==0:
            b+=1
    if b%2==1:
        açık.append(a)
print(len(açık),"dolap kapağı açık kalır. Açık kalanların numaraları:")
print(açık)
```

**Comment by Talla04** [comment]
```python
def toggle_door(door_status):
    return not door_status


def main():
    num_doors = 100
    doors = [False] * num_doors  # Başlangıçta tüm kapaklar kapalı

    for i in range(1, num_doors + 1):
        for j in range(i - 1, num_doors, i):
            doors[j] = toggle_door(doors[j])

    open_doors = [i + 1 for i, door in enumerate(doors) if door]

    print("Açık kalan kapak sayısı:", len(open_doors))
    print("Açık kalan kapak numaraları:", open_doors)


if __name__ == "__main__":
    main()
```
